### PR TITLE
implement code-review

### PR DESCRIPTION
**Pull request by techbrothers3**

### CODE REVIEW

File: Client/src/config.js [hunk 1]
```diff
@@ -0,0 +1,5 @@
+const config = {
+    BASE_URL: "http://localhost:5000",
+};
+
+export default config;
```
Comment (mdshoeb01) [inline Client/src/config.js]: add EOF line

File: Server/controller/addNewBlog.js [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+      session.commitTransaction();
+    } catch (e) {
+      logger.info(JSON.stringify({"statusCode":500, "message":e}));
+      return res.status(500).json({ message: e });
+    }
+    logger.info(JSON.stringify({"statusCode":200,"responseBody":newBlog}));
+    return res.status(200).json({ newBlog });
+  };
+  
+module.exports = addNewBlog;
```
Comment (mdshoeb01) [inline Server/controller/addNewBlog.js]: add EOF line

File: Server/controller/addNewBlog.js [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+    const currentDate = new Date();
+    const newBlog = new Blog({
+      title,
+      description,
+      date: currentDate,
+    });
+    try {
+      await newBlog.save();
+    } catch (e) {
+      logger.error(e);
```
Comment (mdshoeb01) [inline Server/controller/addNewBlog.js:20]: log response status code

File: Server/controller/addNewBlog.js [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+      return res.status(500).json({message: e.message})
+    }
+    try {
+      const session = await mongoose.startSession();
+      session.startTransaction();
+      await newBlog.save(session);
```
Comment (mdshoeb01) [inline Server/controller/addNewBlog.js]: why save the blog again, when it is already saved in Line:14

File: Server/controller/fetchListOfBlogs.js [hunk 5]
```diff
@@ -0,0 +1,19 @@
+const mongoose = require("mongoose");
+const Blog = require("../model/Blog");
+const logger = require("../logger");
+const fetchListOfBlogs = async (req, res,next) => {
+    let blogList;
+    try {
+      blogList = await Blog.find();
+    } catch (e) {
+       //Return a 500 status code if the app cannot connect to db
```
Comment (mdshoeb01) [inline Server/controller/fetchListOfBlogs.js:9]: logger response with statuscode

File: Server/controller/fetchListOfBlogs.js [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+       res.status(500).json({message: "Internal server error cannot connect to db"})
+    }
+    if (!blogList) {
+      return res.status(404).json({ message: "no blogs found" });
+    }
+    logger.info(JSON.stringify({"responseBody": blogList, "statusCode": 200}));
+    res.status(200).json({ blogList });
+  };
+
+module.exports = fetchListOfBlogs
```
Comment (mdshoeb01) [inline Server/controller/fetchListOfBlogs.js]: add EOF line

File: Server/controller/health-controller.js [hunk 7]
```diff
@@ -0,0 +1,9 @@
+const express = require('express');
+
+const app = express();
+
+const healthCheck = (req, res) => {
+    res.status(200).send();
+}
+
+module.exports = {healthCheck};
```
Comment (mdshoeb01) [inline Server/controller/health-controller.js]: add EOF line

File: Server/controller/updateBlog.js [hunk 8]
```diff
@@ -0,0 +1,30 @@
...
+      description,
+    });
+  } catch (e) {
+    logger.error(e);
+    logger.info(JSON.stringify({ "statusCode": 500 }));
+    return res
+      .status(500)
+      .json({ message: "something went wrong while updating pls try again" });
+  }
+  if (!currentBlogToUpdate) {
```
Comment (mdshoeb01) [inline Server/controller/updateBlog.js]: In which scenario this line of code will be executed?

File: Server/db/index.js [hunk 9]
```diff
@@ -1,11 +1,17 @@
...
-
+const express = require("express");
+const { username, password } = require("./config.js");
+const logger = require("../logger.js");
+const app = express();
 mongoose.set("strictQuery", false);
 
 mongoose
   .connect(
     `mongodb+srv://${username}:${password}@mydb.nvmdb.mongodb.net/?retryWrites=true&w=majority&appName=myDb`
```
Comment (mdshoeb01) [inline Server/db/index.js]: add mongo request timeouts.

File: Server/index.js [hunk 10]
```diff
@@ -1,15 +1,27 @@
...
+
 app.use("/api/blogs", blogRouter);
 
-app.use("/api", (req, res) => {
-  res.send("hello world");
-});
+app.use("/health-check", healthRouter);
+
 
-app.listen(5000, () => console.log(`App is running at port 5000...`));
```
Comment (mdshoeb01) [inline Server/index.js:15]: make port value configurable

File: Server/logger.js [hunk 11]
```diff
@@ -0,0 +1,32 @@
...
+      new winston.transports.File({ filename: "logs/error.log", level: "error" }),
+  
+      // Log to the console for debugging during development
+      new winston.transports.Console({
+        format: combine(colorize(), logFormat),
+      }),
+    ],
+  });
+  
+  module.exports = logger;
```
Comment (mdshoeb01) [inline Server/logger.js]: add EOF line

File: Server/model/Blog.js [hunk 12]
```diff
@@ -4,15 +4,34 @@ const Schema = mongoose.Schema;
...
+            message: 'Title cannot be empty',
+        }        
     },
     description: {
-        type: String
+        type: String,
+        required:[true, 'description is Required'],
+        trim: true,
+        validate: {
+            validator: function(value) {
```
Comment (mdshoeb01) [inline Server/model/Blog.js:22]: what is the http status code return when validations fail?

File: Server/model/Blog.js [hunk 13]
```diff
@@ -4,15 +4,34 @@ const Schema = mongoose.Schema;
...
         type: Date,
         default: Date.now
     }
-})
+    
+},
+{ versionKey: false }
+)
 
 module.exports = mongoose.model("Blog",blogSchema);
```
Comment (mdshoeb01) [inline Server/model/Blog.js]: add EOF line

File: Server/route/health-route.js [hunk 14]
```diff
@@ -0,0 +1,9 @@
+const express = require('express');
+const healthRouter = express.Router();
+
+const {healthCheck} = require('../controller/health-controller');
+
+
+healthRouter.get("/", healthCheck);
+
+module.exports = healthRouter;
```
Comment (mdshoeb01) [inline Server/route/health-route.js]: add EOF line

**Review by mdshoeb01 — COMMENTED** [review]
Added review comments, few more points here:
1. add logs folder to .gitignore.
2. Update Readme.md based on discussion.

File: Server/model/Blog.js (hunk 12 above)
Comment (techbrothers3) [reply Server/model/Blog.js:22]: 500 internal server error 
    "message": "Blog validation failed: description: description is Required"

Comment (techbrothers3) [reply Server/model/Blog.js:22]: actually 400 bad req

File: Server/.gitignore [hunk 15]
```diff
@@ -1,2 +1,3 @@
 node_modules
-db/config.js
\ No newline at end of file
+db/config.js
+logs
```
Comment (mdshoeb01) [inline Server/.gitignore]: add EOF line

File: Server/.env [hunk 16]
```diff
@@ -0,0 +1 @@
+PORT=5000
```
Comment (mdshoeb01) [inline Server/.env]: add EOF line

File: Server/controller/addNewBlog.js [hunk 17]
```diff
@@ -14,12 +14,12 @@ const addNewBlog = async (req, res) => {
       await newBlog.save();
     } catch (e) {
       logger.error(e);
-      return res.status(500).json({message: e.message})
+      logger.info(JSON.stringify({"statusCode":500, message: e.message}));
+      return res.status(400).json({message: e.message})
```
Comment (mdshoeb01) [inline Server/controller/addNewBlog.js]: `400` statusCode is returned but logged as `500`

File: Server/controller/addNewBlog.js [hunk 18]
```diff
@@ -14,12 +14,12 @@ const addNewBlog = async (req, res) => {
...
     }
     try {
       const session = await mongoose.startSession();
       session.startTransaction();
-      await newBlog.save(session);
```
Comment (mdshoeb01) [inline Server/controller/addNewBlog.js]: if we are removing this line, lines above and below it are useless, Remove them.

File: Server/controller/deleteBlog.js [hunk 19]
```diff
@@ -15,9 +15,9 @@ const deleteBlog = async (req, res) => {
     return res.status(200).json({ message: "successfully deleted" });
   } catch (e) {
     logger.error(e.message);
-    logger.info(JSON.stringify({ "statusCode": 500 }));
+    logger.info(JSON.stringify({ "statusCode": 400 }));
     return res
-      .status(500)
+      .status(400)
```
Comment (mdshoeb01) [inline Server/controller/deleteBlog.js]: catch block is executed when some error happens, where you have to return Internal server error (500), 400 applies when there is some validation failure.

**Review by mdshoeb01 — COMMENTED** [review]
added few more reviews.

File: Server/controller/updateBlog.js (hunk 8 above)
Comment (mdshoeb01) [reply Server/controller/updateBlog.js]: ?

**Pull request merged**